master/buildbot/steps/artifact.py

```diff
@@ -312,16 +312,19 @@ class CreateArtifactDirectory(ShellCommand):
     descriptionDone="Remote artifact directory created."
 
     def __init__(self,  artifactDirectory=None, artifactServer=None, artifactServerDir=None, artifactServerPort=None,
-                **kwargs):
+                 customArtifactPathRef=None, **kwargs):
         self.artifactDirectory = artifactDirectory
         self.artifactServer = artifactServer
         self.artifactServerDir = artifactServerDir
         self.artifactServerPort = artifactServerPort
+        self.customArtifactPathRef = customArtifactPathRef
         ShellCommand.__init__(self, **kwargs)
 
     def start(self):
         br = self.build.requests[0]
-        if mkdt(br.submittedAt) > ARTIFACT_LOCATION_CHANGE_DATE:
+        if callable(self.customArtifactPathRef):
+            artifactPath = self.customArtifactPathRef(self.build)
+        elif mkdt(br.submittedAt) > ARTIFACT_LOCATION_CHANGE_DATE:
```

Comment:
I completely forgot about this `ARTIFACT_LOCATION_CHANGE_DATE`, it was implemented when we were changing artifact locations.

It can be removed (you can even delete the constant), since `mkdt(br.submittedAt) > ARTIFACT_LOCATION_CHANGE_DATE` will always be `True`

Reply:
I've removed this constant and dead code (last else section). I've adjusted tests for this code.